go/libkb/skb.go

```diff
@@ -237,11 +237,11 @@ func (s *SKB) UnlockSecretKey(m MetaContext, passphrase string, tsec Triplesec,
 	}
 	var unlocked []byte
 
-	switch s.Priv.Encryption {
-	case 0:
+	switch {
+	case s.Priv.Encryption == 0:
 		m.CDebugf("case: Unlocked")
 		unlocked = s.Priv.Data
-	case int(triplesec.Version):
+	case s.Priv.Encryption > 0 && s.Priv.Encryption < LKSecVersion:
```

Comment:
Not really sure how to do this. This change reserves 1 through 99 for triplesec versions, and should be OK because newer versions of triplesec can decrypt older versions as well. Not sure what happens if the default ClientTriplesecVersion is 4, but the user provides a version 3 ciphertext. This would cache a version 4 stream. Is that okay as long as the client can decrypt that later?